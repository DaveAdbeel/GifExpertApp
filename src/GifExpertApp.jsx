import { useState } from "react"
import AddCategory from "./components/AddCategory"
import GifGrid from "./components/GifGrid";


const GifExpertApp = () => {
    const [categories, setCategories] = useState([]); 
    const onAddCategory = (newCategory) => {
        if (!categories.includes(newCategory))
            setCategories([newCategory, ...categories])
    }
    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onInputSubmit={onAddCategory}/>
            <ol aria-label="gif-list">
                {categories.map( (category, key) => <GifGrid  key={key} category={category}/>)}
            </ol>
        </>
    )
}

export default GifExpertApp