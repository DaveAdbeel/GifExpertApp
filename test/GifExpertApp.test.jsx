/* eslint-disable no-undef */

import { screen, render } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe("Pruebas en nuestra app <GidExpertApp/>", () => {
    const titleApp = "GifExpertApp";
    test("Al montar la App tiene que hacer match con el snapshot", (done) => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        done();
    });

    test("Al montar la App debe tener el titulo 'GifExpertApp' ", (done) => {
        render(<GifExpertApp />);
        expect(screen.getByText(titleApp));
        done();
    });

    test("Al montar la App debe tener el formulario para buscar gifs", (done) => {
        render(<GifExpertApp />);
        expect(screen.getByRole("form", { name: "form" }));
        done();
    });
   
    test("Al montar la App debe haber una lista vacia, esperando por los gifs", (done) => {
        render(<GifExpertApp />)
        expect(screen.getByRole("list", {name: "gif-list"}));
        done();
    });
});
