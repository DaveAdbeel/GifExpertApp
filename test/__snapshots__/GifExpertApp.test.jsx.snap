// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en nuestra app <GidExpertApp/> Al montar la App tiene que hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      aria-label="gifs-search-bar"
      placeholder="Busca tus gifs"
      type="text"
      value=""
    />
  </form>
  <ol
    aria-label="gif-list"
  />
</div>
`;
